src/transformers/modeling_tf_utils.py

```diff
@@ -706,7 +833,20 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         # 'by_name' allow us to do transfer learning by skipping/adding layers
         # see https://github.com/tensorflow/tensorflow/blob/00fad90125b18b80fe054de1055770cfb8fe4ba3/tensorflow/python/keras/engine/network.py#L1339-L1357
         try:
-            load_tf_weights(model, resolved_archive_file)
+            # Temporary fix in order to detect if the loaded model adopts the new TF code design or not.
+            # This will be removed once all the TF models will be updated to the new design
+            if model.base_model_prefix in ["bert"]:
```

Comment:
IMO, it would be much nicer and cleaner to just have a single new "load_tf_weights" function here. I'm more than willing to accept a bad weight parameter naming to avoid having to add string replacement hacks in the load_function and if-statements like the one above.

Reply:
This is because I didn't updated the other models. This will be removed only once I will have updated all the others.